bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java

```diff
@@ -64,6 +64,11 @@ public String getThankYouMessage() {
         return thankyouMessage;
     }
 
+    @Override
+    public String getAdaptiveFormVersion() {
```

Comment:
this should be picked up from storage (jcr here).

Reply:
No. Any changes in spec would mean a new version of form container. Hence the value needs to be hard coded

Reply:
spec follows semantic versioning. the changes in minor/patch versions should not need another version of core component.

Reply:
That depends on the use-case. if we need to a new component, we would add it, else we could add the changes in the same component.